conans/model/env.py

```diff
@@ -0,0 +1,184 @@
+from collections import OrderedDict
+from conans.util.log import logger
+import os
+import re
+
+
+class EnvValues(object):
+    """ Object to represent the introduced env values entered by the user
+    with the -e or profiles etc.
+        self._data is a dictionary with: {(package, var): value}. "package" can be None if the var is global.
+    """
+
+    def __init__(self):
+        self._data = {}
+
+    @property
+    def data(self):
+        return self._data
+
+    @property
+    def sorted_data(self):
+        # Python 3 can't compare None with strings, so if None we order just with the var name
+        return sorted(self._data.items(), key=lambda x: "%s_%s" % x if x[0] else x[1])
+
+    def add(self, name, value, package=None):
+        # It prioritizes the data already introduced
+        if (package, name) not in self._data:
+            self._data[(package, name)] = value
+
+    def global_values(self):
+        '''return [(name, value), (name, value)] for global env variables'''
+        return [(name, value) for (pname, name), value in self.sorted_data
+                if pname is None]
+
+    def package_values(self, package):
+        """return the environment variables that apply for a given package name:
+        [(name, value), (name, value)] for the specified package"""
+        assert(package is not None)
+        return [(name, value) for (pname, name), value in self.sorted_data
+                if pname == package]
+
+    def all_package_values(self):
+        return sorted([("%s:%s" % (pname, name), value) for (pname, name), value in self.data.items()
+                       if pname])
+
+    def update(self, env_obj):
+        """accepts other EnvValues object or DepsEnvInfo
+           it prioritize the values that are already at self._data
+        """
+        if env_obj:
+            if isinstance(env_obj, EnvValues):
+                for (package_name, name), value in env_obj._data.items():
+                    self.add(name, value, package_name)
+            else:  # DepsEnvInfo
+                for (name, value) in env_obj.vars.items():
+                    name = name.upper() if name.lower() == "path" else name
+                    if isinstance(value, list):
+                        value = os.pathsep.join(value)
```

Comment:
Hurray! Comments in code always a good idea, especially for people that are not "the present yourself"